### PR TITLE
Posters return to your hand when cut down/not hung up, abductor posters more resilient to wirecutters

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Posters now drop into your hand when you cancel the hanging process, or snip them down with wirecutters. If they can't drop into your hands, they go to the floor.

Oh, also, this makes it so that abductor wirecutters (or any wirecutters with a toolspeed better than 0.2 in case of varedits/future additions) are too weak to cut down abductor posters.
## Why It's Good For The Game

Posters getting dropped to the ground when cancelling the hanging process has always really annoyed me and I want it to not happen anymore.

If you want to get those unsightly abductor posters out of your department you'll have to loot the abductor scientist first (or just take the wall down but that's admitting defeat).
## Changelog
:cl: Rhials
qol: Posters now return to your hand when the hanging process is cancelled, or they are snipped down. 
qol: Abductor posters can now only be cut down with abductor-tier wirecutters.
/:cl:

### CODE REVIEW

**Comment by DaCoolBoss** [comment]
Can the posters still be burned off or ripped? The situation I'm worried about is an abductor smashing an APC and postering over it to make it very difficult to repair.

**Comment by Rhials** [comment]
> Can the posters still be burned off or ripped? The situation I'm worried about is an abductor smashing an APC and postering over it to make it very difficult to repair.

They cannot be ripped down by non-abductors. They are still burnable.

**Comment by san7890** [comment]
actually wait abductor posters don't do anything i thought it was like traitor posters that demoralize you. still feels weird but whatever

File: code/game/objects/effects/posters/poster.dm [hunk 1]
```diff
@@ -207,11 +207,12 @@
 		return FALSE
 	return !user.gloves || !(user.gloves.body_parts_covered & HANDS) || HAS_TRAIT(user, TRAIT_FINGERPRINT_PASSTHROUGH) || HAS_TRAIT(user.gloves, TRAIT_FINGERPRINT_PASSTHROUGH)
 
-/obj/structure/sign/poster/proc/roll_and_drop(atom/location)
+/obj/structure/sign/poster/proc/roll_and_drop(atom/location, mob/user)
 	pixel_x = 0
 	pixel_y = 0
 	var/obj/item/poster/rolled_poster = new poster_item_type(location, src) // /obj/structure/sign/poster/wanted/roll_and_drop() has some snowflake handling due to icon memes, if you make a major change to this, don't forget to update it too. <3
-	forceMove(rolled_poster)
+	if(!user || !user.put_in_hands(rolled_poster))
```
Comment (Jacquerel) [inline code/game/objects/effects/posters/poster.dm:214]:
```suggestion
	if(isnull(user) || !user.put_in_hands(rolled_poster))
```

Comment (Jacquerel) [reply code/game/objects/effects/posters/poster.dm:214]:
```suggestion
	if(!user?.put_in_hands(rolled_poster))
```
Alternately

File: code/modules/antagonists/abductor/equipment/gear/abductor_posters.dm [hunk 2]
```diff
@@ -17,6 +17,12 @@
 		return
 	return ..()
 
+/obj/structure/sign/poster/abductor/attackby(obj/item/tool, mob/user, params)
+	if(tool.toolspeed >= 0.2)
+		to_chat(user, span_warning("Your [tool.name] is too weak to cut through the superior adhesive on the [name]!"))
```
Comment (Jacquerel) [inline code/modules/antagonists/abductor/equipment/gear/abductor_posters.dm:22]:
```suggestion
		balloon_alert(user, "tool too weak!")
```
This doesn't really seem important enough to need to stick around in the chat log.

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Some real minor nitpicks.

**Pull request merged**